cpp/src/arrow/dataset/partition.h

```diff
@@ -76,7 +78,8 @@ class ARROW_DS_EXPORT Partitioning {
   /// \brief Parse a path into a partition expression
   virtual Result<compute::Expression> Parse(const std::string& path) const = 0;
 
-  virtual Result<std::string> Format(const compute::Expression& expr) const = 0;
+  virtual Result<std::pair<std::string, std::string>> Format(
+      const compute::Expression& expr) const = 0;
```

Comment:
The new signature isn't immediately understandable (what does the pair represent?).
Can you instead make it return a locally-defined struct e.g.:
```suggestion
  struct SomeStruct { std::string foo, bar };
  virtual Result<SomeStruct> Format(const compute::Expression& expr) const = 0;
```

Reply:
Changed function return signature to structure. I am not very sure of the name of the struct use, does that sound okay?